bitcoin/embedded/src/main.rs

```diff
@@ -43,8 +43,8 @@ fn main() -> ! {
     let secp = Secp256k1::preallocated_new(&mut buf_ful).unwrap();
 
     // Derive address
-    let pubkey = pk.public_key(&secp);
-    let address = Address::p2wpkh(&pubkey, Network::Bitcoin).unwrap();
+    let pubkey = pk.public_key(&secp).try_into().unwrap();
```

Comment:
In bc1qpx9t9pzzl4qsydmhyt6ctrxxjd4ep549np9993:

I think that `PrivateKey` should have a dedicated constructor for getting a compressed public key. Maybe we should have a `CompressedPrivateKey` type?

Reply:
Oops, in a92d49fe33d8c304b4453e2d5dc73c5dfe577ee1 I meant. I didn't look at what my copy-interesting-strings-from-terminal script was giving me :P.

Reply:
> Maybe we should have a `CompressedPrivateKey` type?

I did think about that but given `PrivateKey` is effectively `WifKey` it doesn't seem to make much sense. Should applications reject uncompressed private keys when parsing?

Reply:
Yeah, I think you're right, it doesn't really make sense (or rather, it's not obvious what the behavior would be and it has very limited use cases).

Personally I think the most useful thing would be for the type to parse any WIF key but just throw away the compressedness byte on the assumption that if it's set, it's a mistake or a result of some legacy system being interfaced with something more modern. But that's a very dangerous thing to do implicitly because it results in unexpected addresses.

Reply:
Well, it'd make it impossible to implement e.g. a tool that sweeps legacy addresses. I think that's enough reason not to do it.